test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NetCoreProjectTestCase.cs

```diff
@@ -258,5 +258,100 @@ public static IEnumerable<object[]> GetNetCoreTemplates()
         {
             yield return new object[] { ProjectTemplate.NetStandardClassLib };
         }
+
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetNetCoreProjects))]
+        public void InstallPackageToNetCoreProjectFromUI(ProjectTemplate projectTemplate, string projectName, string packageName, string packageVersion)
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+            var dte = VisualStudio.Dte;
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution();
+
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, projectName);
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, XunitLogger);
+            var nugetTestService = GetNuGetTestService();
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(packageName, packageVersion);
+
+            // Assert
+            VisualStudio.AssertNuGetOutputDoesNotHaveErrors();
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, packageName, packageVersion, XunitLogger);
+        }
+
+        public static IEnumerable<object[]> GetNetCoreProjects()
+        {
+            yield return new object[] { ProjectTemplate.NetStandardClassLib, "NetStandardClassLib", "NetStandard.Library", "2.0.1" };
```

Comment:
I thought the .NET SDK automatically adds `NetStandard.Library` to `netstandard` TFMs. If that's right, then using `NetStandard.Library` as a package to install in the project is not a good choice. `Newtonsoft.Json` version `13.0.1` is probably a better choice, but ideally it's a package that's already installed by other tests, to minimize the number of network calls tests will make (in the past we had flaky tests when the CI agent's network stopped working temporarily, although we might have had an HTTP retry bug as well).

My preference would actually be to install a generated package from a local filesystem source, because I wish none of our tests made any network requests, but I'm not sure we have any examples in Apex tests already that do this. I think it would be great to have a test fixture that create the test package(s) one time, and multiple tests share them, to avoid having tests re-create similar test packages over and over in a single test execution.

Reply:
Network requests are made via `System.Net.Http.HttpClient`, which is part of the .NET Base Class Library, so in my opinion, it's the .NET team's responsibility to test. NuGet has unit tests to make sure that NuGet.Protocol uses HttpClient correctly (both via `HttpClient` with a mock `HttpMessageHandler`, and also via HTTP server's listening on `localhost`, so it doesn't actually use the network). I don't see it as something that provides us with much safety in a Visual Studio integration test.

It's been a long time since I've seen a flaky network cause a test failure to nuget.org, but I'd still like to minimize risk of unreliable tests preventing pull requests from being merged.

Reply:
Also, maybe this is similar to the discussion we started with the previous Apex test you added, `NuGet.Tests.Apex` runs on every pull request. If you'd like to write tests that have a lot more duplicated, exhaustive tests, even though we believe it's testing the same implementation as what we already have tests for, then we can investigate [adding a new test project that runs once a day/week](https://github.com/NuGet/Client.Engineering/issues/2020).

Reply:
> Network requests are made via `System.Net.Http.HttpClient`, which is part of the .NET Base Class Library, so in my opinion, it's the .NET team's responsibility to test. NuGet has unit tests to make sure that NuGet.Protocol uses HttpClient correctly (both via `HttpClient` with a mock `HttpMessageHandler`, and also via HTTP server's listening on `localhost`, so it doesn't actually use the network). I don't see it as something that provides us with much safety in a Visual Studio integration test.
> 
> It's been a long time since I've seen a flaky network cause a test failure to nuget.org, but I'd still like to minimize risk of unreliable tests preventing pull requests from being merged.

I see. As you mentioned above, should we use a package (e.g. Newtonsoft.Json) that's already installed by other tests in `NuGet.Tests.Apex` instead?

Reply:
> Also, maybe this is similar to the discussion we started with the previous Apex test you added, `NuGet.Tests.Apex` runs on every pull request. If you'd like to write tests that have a lot more duplicated, exhaustive tests, even though we believe it's testing the same implementation as what we already have tests for, then we can investigate [adding a new test project that runs once a day/week](https://github.com/NuGet/Client.Engineering/issues/2020).

Yes, we'd like to write tests which can replace at least part of manual testing we do every day. We plan to add more and more projects gradually to our test cases after the test pipeline created. Before [the task](https://github.com/NuGet/Client.Engineering/issues/2020) is done, we just want to pick one project to write test case for one scenario and merge it to NuGet build, wishing this will bring minimum impact to your work. Does the plan make sense?